### PR TITLE
FaceID: registration + verification + assets

### PR DESCRIPTION
**Pull request by bielikb**
# Changes

* New registration step - `FaceID scan + verification`
* New verification step  - `FaceID verification` - currently this feature is commented in `MainViewController` - please see my comment in this PR

# Flow
`FaceCaptureCoordinator` is the most visible part of the new `FaceIDCapture` component
`FaceCaptureCoordinator` takes care of two different use cases:
```
enum FaceIDUseCase {
    case registerFace
    case verifyFace
}
```

and it supports these flow steps described by `FaceCaptureCoordinatorStep` enum:
```
    enum FaceCaptureCoordinatorStep {
        case initialised
        case onboarding
        case faceCapture
        case faceVerification
        case completion
    }
```

`FaceCaptureCoordinator` is instantiated as needed in `MainViewController`. 
`MainViewController` observes `StartFaceIDRegistrationNotification` that fires the `FaceCaptureCoordinator` with `.registerFace` use case.

# FaceID Service:

 Instantiation and basic configuration of DOT framework is handled by `FaceID` interface. It also reads the value from `Firebase` s config.

* `FaceIDCapture` is a wrapper and delegate  for `FaceCapture` controller. It handles all important logic around face id capture
* `LivenessStepCoordinator` is a wrapper and handler for liveness check
* `FaceIDValidator` verifies how does the given sample(s) match the reference template
* `FaceIDStorage` is a storage for reference template = this is cached and stored once the user gets through the registratrion flow

### CODE REVIEW

File: Covid/MainScreen/MainViewController.swift [hunk 1]
```diff
@@ -66,6 +71,9 @@ final class MainViewController: UIViewController {
         super.viewDidAppear(animated)
         registerUser()
 
+//        after(.seconds(1)) { [weak self] in
```
Comment (bielikb) [inline Covid/MainScreen/MainViewController.swift]: intermediate code == by uncommenting this code, verification FaceID flow can be tested

File: Covid/Extensions/General.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (anonymcek) [inline Covid/Extensions/General.swift:23]: pohadz do tych novych files MIT licku ako je v ostatnych..

File: Covid/Services/FaceID/FaceIDStorage.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @SwiftyUserDefault(keyPath: \.referenceFace, options: [.cached, .observed])
+    var referenceFaceData: [Int8]?
+
+    func saveReferenceFace(_ faceCaptureImage: FaceCaptureImage) {
+        referenceFaceData = faceCaptureImage.faceTemplate?.data
+    }
+}
+
+extension DefaultsKeys {
+    var referenceFace: DefaultsKey<[Int8]?> { .init("dot-face") }
```
Comment (anonymcek) [inline Covid/Services/FaceID/FaceIDStorage.swift]: ja som vsetky definoval na jednom mieste, nech to mame pokope, ale tak je to asi jedno

File: Covid/en.lproj/Main.strings [hunk 4]
```diff
@@ -0,0 +1,213 @@
+
```
Comment (anonymcek) [inline Covid/en.lproj/Main.strings:1]: to len aby sme mali? ked budu texty prelozime?

File: Covid/Services/FaceID/FaceIDStepVerifier.swift [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        debugPrint(#function)
+
+        guard let status = DOTHandler.authorizeCamera(onRequestAccess: { [weak self] in
+            self?.checkCameraPermission(status: $0, controller: controller)
+        }) else { return }
+
+        checkCameraPermission(status: status.authorizationStatus, controller: controller)
+    }
+
+    func livenessCheckDidLoad(_ controller: LivenessCheckController) {
```
Comment (anonymcek) [inline Covid/Services/FaceID/FaceIDStepVerifier.swift:179]: tieto prazdne veci asi mozu prec, ci?

File: Covid/Services/FaceID/FaceIDCapture.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    func faceCaptureDidLoad(_ controller: FaceCaptureController) {
+        debugPrint(#function)
+    }
+
+    func faceCaptureDidAppear(_ controller: FaceCaptureController) {
+        controller.requestFaceCapture()
+    }
+
+    func faceCaptureWillAppear(_ controller: FaceCaptureController) {
```
Comment (anonymcek) [inline Covid/Services/FaceID/FaceIDCapture.swift:91]: aj tuu

File: Covid/Quarantine/FaceID/FaceCaptureCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        let result = faceIdValidator.validateSegmentImagesToReferenceTemplate(segmentImages)
+        switch result {
+        case .success:
+            print("verify: success")
+        default:
+            break
+        }
+    }
+
+    func livenessdidFailedWithEyesNotDetected(_ step: LivenessStepCoordinator) {
```
Comment (anonymcek) [inline Covid/Quarantine/FaceID/FaceCaptureCoordinator.swift:219]: toto neva ze nic nerobime?

**Review by anonymcek — CHANGES_REQUESTED** [review]
drobnostky, este to rano otestim

File: Covid/Quarantine/FaceID/FaceCaptureCoordinator.swift (hunk 7 above)
Comment (bielikb) [reply Covid/Quarantine/FaceID/FaceCaptureCoordinator.swift:219]: good catch, osetril som alertom, ktory pouzivame pri scane

File: Covid/Services/FaceID/FaceIDStepVerifier.swift (hunk 5 above)
Comment (bielikb) [reply Covid/Services/FaceID/FaceIDStepVerifier.swift:179]: tieto su celkom potrebne pre development, ukryl som ich za debug compile flag

File: Covid/Services/FaceID/FaceIDStorage.swift (hunk 3 above)
Comment (bielikb) [reply Covid/Services/FaceID/FaceIDStorage.swift]: key som hodil k ostatnym. dik

File: Covid/MainScreen/MainViewController.swift (hunk 1 above)
Comment (anonymcek) [reply Covid/MainScreen/MainViewController.swift]: este, nepovolime ten faceid nech to aj chalani vedia otestit?

Comment (bielikb) [reply Covid/MainScreen/MainViewController.swift]: rozmyslal som nad tym. eg pridat moznost do settingsov appky. tzn jednoduchy switch, ktory by vyhadzoval flow na verifikaciu pri starte appky, teda za predpokladu, ze faceID template je po ruke == uzivatel presiel faceID registraciou

Comment (anonymcek) [reply Covid/MainScreen/MainViewController.swift]: a sak nekomplikujme si to, nech aj ti backendaci nieco porobia

**Review by anonymcek — APPROVED** [review]

**Pull request merged**